From 92571a8cf7fdd2c3dcf3cfc61955dfaa553e7f9d Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-92571a8cf7-20210512T094803Z-avarab@gmail.com>
In-Reply-To: <xmqqr1idn41e.fsf@gitster.g>
References: <xmqqr1idn41e.fsf@gitster.g>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 10 May 2021 11:02:05 +0200
Subject: [PATCH] Makefile: make PERL_DEFINES recursively expanded
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since 07d90eadb50 (Makefile: add Perl runtime prefix support,
2018-04-10) PERL_DEFINES has been a simply-expanded variable, let's
make it recursively expanded instead.

This change doesn't matter for the correctness of the logic. Whether
we used simply-expanded or recursively expanded didn't change what we
wrote out in GIT-PERL-DEFINES, but being consistent with other rules
makes this easier to understand.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

This is a trivial improvement-on-top for my ab/perl-makefile-cleanup
already in "next". I'd sent a v2 of that in [1] not seeing it was
merged down already, this incremental patch replaces that and results
in the same end-state.

1. https://lore.kernel.org/git/cover-0.6-00000000000-20210510T104937Z-avarab@gmail.com/

 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b8d6b31305..0705fc2d3f 100644
--- a/Makefile
+++ b/Makefile
@@ -2271,7 +2271,7 @@ perl_localedir_SQ = $(localedir_SQ)
 
 ifndef NO_PERL
 PERL_HEADER_TEMPLATE = perl/header_templates/fixed_prefix.template.pl
-PERL_DEFINES :=
+PERL_DEFINES =
 PERL_DEFINES += $(PERL_PATH_SQ)
 PERL_DEFINES += $(PERLLIB_EXTRA_SQ)
 PERL_DEFINES += $(perllibdir_SQ)
-- 
2.31.1.909.g789bb6d90e

